.github/CONTRIBUTING.md

```diff
@@ -25,22 +25,57 @@ You'll see exactly what is sent, be able to add your comments, and then send it.
+
+* Make changes on a **separate branch** with a meaningful name, not on the _master_ branch or the _dev_ branch. This is commonly known as *feature branch workflow*. You may then send your changes as a pull request (PR) on GitHub.
+* Please **test** (compile and run) your code before submitting changes! Ideally, provide test feedback in the PR description. Untested code will **not** be merged!
 * Respond if someone requests changes or otherwise raises issues about your PRs.
-* Send PRs that only cover one specific issue/solution/bug. Do not send PRs that are huge and consist of multiple independent solutions.
 * Try to figure out yourself why builds on our CI fail.
+* Make sure your PR is **up-to-date** with the rest of the code. Often, a simple click on "Update branch" will do the job, but if not, you must *rebase* your branch on the `dev` branch manually and resolve the conflicts on your own. You can find help [on the wiki](https://github.com/TeamNewPipe/NewPipe/wiki/How-to-merge-a-PR). Doing this makes the maintainers' job way easier.
+
+## IDE setup & building the app
+
+### Basic setup
+
+NewPipe is developed using [Android Studio](https://developer.android.com/studio/). Learn more about how to install it and how it works in the [official documentation](https://developer.android.com/studio/intro). In particular, make sure you have accepted Android Studio's SDK licences. Once Android Studio is ready, setting up the NewPipe project is fairly simple:
+- Clone the NewPipe repository with `git clone https://github.com/TeamNewPipe/NewPipe.git` (or use use the link from your own fork, if you then want to open a PR).
+- Open the folder you just cloned with Android Studio.
+- Build and run just like you would do with any other app, with the green triangle in the top bar.
+
+### checkStyle setup
+
+The [checkStyle](https://github.com/checkstyle/checkstyle) plugin verifies that Java code abides by the project style. It runs automatically each time you build the project, though to be able to view errors directly in the editor, instead of having to skim through the build output, you can install an Android Studio plugin:
+- Go to `File -> Settings -> Plugins`, search for `checkstyle` and install `CheckStyle-IDEA`.
+- Go to `File -> Settings -> Tools -> Checkstyle`.
+- Add NewPipe's configuration file by clicking the `+` in the right toolbar of the "Configuration File" list.
+- Under the "Use a local Checkstyle file" bullet, click on `Browse` and pick the file named `checkstyle.xml` in the project's root folder.
+- Enable "Store relative to project location" so that moving the directory around does not create issues.
+- Insert a description in the top bar, then click `Next` and then `Finish`.
+- Activate the configuration file you just added by enabling the checkbox on the left.
+- Click `Ok` and you are done.
+
+An alternative plugin you can choose for the same job is [SonarLint](https://www.sonarlint.org/intellij), which works for both Java and Kotlin.
```

Comment:
I wouldn't say that's an alternative, but more like an addition 😄

Reply:
Uhm, isn't it a replacement for the two plugins for checkstyle and ktlint?

Reply:
no. SonarLint tuns a in-depth code analysis and thus helps to find potential bugs or performance problems.

Reply:
Oh ok